sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java

```diff
@@ -1890,17 +1887,21 @@ public void populateDisplayData(DisplayData.Builder builder) {
     private final DisplayData.ItemSpec<? extends Class<?>> fnDisplayData;
     private final SerializableFunction<? super K, Integer> hotKeyFanout;
 
-    private PerKeyWithHotKeyFanout(String name,
+    private PerKeyWithHotKeyFanout(
         PerKeyCombineFn<? super K, ? super InputT, ?, OutputT> fn,
         DisplayData.ItemSpec<? extends Class<?>> fnDisplayData,
         SerializableFunction<? super K, Integer> hotKeyFanout) {
-      super(name);
       this.fn = fn;
       this.fnDisplayData = fnDisplayData;
       this.hotKeyFanout = hotKeyFanout;
     }
 
     @Override
+    protected String getKindString() {
+      return String.format("Combine.perKeyWithFanout(%s)", NameUtils.approximateSimpleName(fn));
```

Comment:
Just checking - is this inlining the existing behavior? It seems bogus because `Combine.perKeyWithFanout` is not a valid expression even though it is phrased as though it is. If it matches existing behavior, it should stay as you have it. Otherwise it should be something else, maybe just capitalized to match the `ParDo` kind string (which appears to be phrased in sort of a Python/Haskell style).

Small potatoes, though, since it should be replaced by the various runner/fn api URNs for mechanical purposes and display data for UI purposes.